### DISCUSSION TITLE
select() function (select from cursor pos 1 → pos 2) ?

### DISCUSSION THREAD

**Discussion by luckman212** [SilentVoid13/Templater#606] [Ideas & Feature Requests]
Hi
was wondering if someone knows a way to have Templater load a template with a certain section of text pre-selected, e.g.

```
> [!Info]
> **N.B.** Something <% tp.selection.start() %>yada yada<% tp.selection.end() %>
```

...and have `yada yada` be selected upon creation?

**Comment by AB1908** [comment]
Would likely need to use Obs API for this but what's the use case?

**Comment by luckman212** [comment]
the use case could be something like a form letter, e.g.
```
Dear <% tp.selection.start() %>Person_name<% tp.selection.end() %>,
blah blah...
```
so you can just insert the template and start typing immediately to overwrite the default name. Yes I understand you could do
```
Dear <% tp.file.cursor() %>,
blah blah
```
But that doesn't allow for a "default" value and isn't quite as nice to look at when you're editing the template itself...

**Reply by AB1908** [reply]
How about using a `tp.system.prompt` to overwrite a default value? Could add some conditional logic to handle when no input in prompt.
